parlai/agents/bert_qa/bert_qa.py

```diff
@@ -0,0 +1,421 @@
+        #                                       input_ids)
+        #     predictions.append(prediction)
+
+        # return Output(predictions)
+
+    def eval_step(self, batch):
+
+        segment_ids = torch.zeros(batch['text_vec'].size(),
+                                  dtype=torch.long).to(self.device)
+        input_mask = batch['text_vec'].clone()
+        input_mask[input_mask != self.dict.pad_idx] = 1
+
+        with torch.no_grad():
+            b_start_logits, b_end_logits = self.model(batch['text_vec'],
+                                                      segment_ids, input_mask)
+
+        predictions = []
+        for start_logits, end_logits, input_ids in zip(b_start_logits,
+                                                       b_end_logits,
+                                                       batch['text_vec']):
+            prediction = self._get_prediction(start_logits, end_logits,
+                                              input_ids)
+            predictions.append(prediction)
+
+        return Output(predictions)
+
+    def share(self):
+        """Share model parameters."""
+        shared = super().share()
+        shared['model'] = self.model
+        shared['metrics'] = self.metrics
+        return shared
+
+    @staticmethod
+    def add_cmdline_args(parser):
+        TorchAgent.add_cmdline_args(parser)
+        parser = parser.add_argument_group('BERT Classifier Arguments')
+        parser.add_argument('--type-optimization',
+                            type=str,
+                            default='all_encoder_layers',
+                            choices=[
+                                'additional_layers', 'top_layer',
+                                'top4_layers', 'all_encoder_layers', 'all'
+                            ],
+                            help='which part of the encoders do we optimize '
+                            '(defaults to all layers)')
+        parser.add_argument(
+            "--max_text_length",
+            default=256,
+            type=int,
+            help=
+            "The maximum number of tokens for the text. Datapoints with a text"
+            "longer than this will be ignored.")
+        parser.add_argument(
+            "--n_best_size",
+            default=20,
+            type=int,
+            help=
+            "The total number of n-best predictions to generate in the nbest_predictions.json "
+            "output file.",
+        )
+        parser.add_argument(
+            "--bert_model",
+            default='bert-base-cased',
+            type=str,
+            help=
+            "Bert pre-trained model selected in the list: bert-base-uncased, "
+            "bert-large-uncased, bert-base-cased, bert-large-cased, bert-base-multilingual-uncased, "
+            "bert-base-multilingual-cased, bert-base-chinese.",
+        )
+        parser.add_argument(
+            "--warmup_proportion",
```

Comment:
--warmup-updates is supposed to be this

Reply:
--warmup-updates is an integer. 
--warmup_proportion is a double in [0,1]